utilities/virt.py

```
@@ -2146,26 +2146,63 @@ def vm_instance_from_template(
+        compact_cluster: If True, relocate virt-api pods before drain.
+    """
+    if compact_cluster:
+        for pod in utilities.infra.get_pods(
+            client=admin_client,
+            namespace=hco_namespace,
+            label=f"{Pod.ApiGroup.KUBEVIRT_IO}={VIRT_API}",
+        ):
+            if pod.node.name == node.name:
+                LOGGER.info(
+                    f"Compact cluster: virt-api pod {pod.name} on {node.name}, "
+                    "deleting before drain to avoid webhook race"
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)